contrib/babelfishpg_tds/src/backend/tds/tdsbulkload.c

```diff
@@ -794,12 +801,15 @@ SetBulkLoadRowData(TDSRequestBulkLoad request, StringInfo message)
 						{
 							case TDS_TYPE_TEXT:
 								rowData->columnValues[i] = TdsTypeVarcharToDatum(temp, colmetadata[i].encoding, colmetadata[i].columnTdsType);
+								rowData->isAllocated[i] = true;
 								break;
 							case TDS_TYPE_NTEXT:
 								rowData->columnValues[i] = TdsTypeNCharToDatum(temp);
+								rowData->isAllocated[i] = true;
 								break;
 							case TDS_TYPE_IMAGE:
 								rowData->columnValues[i] = TdsTypeVarbinaryToDatum(temp);
+								rowData->isAllocated[i] = true;
```

Comment:
Common condition between all cases, can we move it out?

Reply:
Moved it out of the condition and added a comment.